packages/std/src/deps.rs

```diff
@@ -1,11 +1,53 @@
-use crate::traits::{Api, Querier, Storage};
+use crate::traits::{Api, QuerierTrait, Storage};
+use crate::Querier;
 
 /// Holds all external dependencies of the contract.
 /// Designed to allow easy dependency injection at runtime.
 /// This cannot be copied or cloned since it would behave differently
 /// for mock storages and a bridge storage in the VM.
-pub struct Deps<S: Storage, A: Api, Q: Querier> {
+pub struct OwnedDeps<S: Storage, A: Api, Q: QuerierTrait> {
     pub storage: S,
     pub api: A,
     pub querier: Q,
 }
+
+pub struct Deps<'a> {
+    pub storage: &'a mut dyn Storage,
+    pub api: &'a dyn Api,
+    pub querier: Querier<'a>,
+}
+
+#[derive(Copy, Clone)]
+pub struct DepsRef<'a> {
```

Comment:
I'd call this `Deps` and the mutable version `DepsMut`. I think those names are better than the "Ref" suffix. This also uses the typical immutable default.

Reply:
Okay, doing that now, then preparing for merge